zebra-chain/src/orchard/shielded_data.rs

```diff
@@ -88,9 +82,34 @@ impl TrustedPreallocate for Action {
 /// [ps] https://zips.z.cash/protocol/nu5.pdf#actionencodingandconsensus
 pub const SPEND_AUTH_SIG_SIZE: u64 = 64;
 
+/// The size of a single AuthorizedAction
+///
+/// Is the size of an `Action` + a `Signature<SpendAuth>`
+pub const AUTHORIZED_ACTION_SIZE: u64 = ACTION_SIZE + SPEND_AUTH_SIG_SIZE;
+
+/// The maximum number of authorized orchard actions in a valid Zcash on-chain transaction V5.
+///
+/// If a transaction contains more actions than can fit in maximally large block, it might be
+/// valid on the network and in the mempool, but it can never be mined into a block. So
+/// rejecting these large edge-case transactions can never break consensus.
+impl TrustedPreallocate for AuthorizedAction {
```

Comment:
Do we need to `impl TrustedPreallocate for AuthorizedAction`?

`impl TrustedPreallocate for AuthorizedAction` allows us to deserialize arrays of `AuthorizedAction`s, but there are no `AuthorizedAction` arrays in V5 transaction serialization. (V5 has separate `Action` and `Signature<SpendAuth>` arrays.)